### ISSUE TITLE
Add Email Client Connection

### ISSUE THREAD

**Issue by JustinPhlegar** [Orasi/Selenium-Java-Core#9]
Add ability to connect to any email client to send and possibly retrieve emails.

**Comment by simpkinscm** [comment]
For retrieving emails, what type of returns are we wanting?  I figure the most useful on would be returning an array of unread emails; but what other type of returns would be good to have?

**Comment by JustinPhlegar** [comment]
@Mattwhooo @jessicamarshall @adamdthomas @waitsavery 

What are your guys thoughts? Just returning an array of emails is enough and let user handle it how he wants? Should there be methods to filter to a specific email and return it?

**Comment by Mattwhooo** [comment]
With retrieving emails I feel like just something that emails that arrived in the past x time would be good as that is likely what you are going to be after.  Sending emails there are way easier ways to do this then with full email client.  We do this with the Rails app to send emails and I've done it with a dot net client before, but I would have to look up the library.  

Retrieving emails seems like some what of a fools errand for integrating in to the core though.  There is too much variability with different email setups and we would likely have to rewrite it everytime we implement.

**Comment by simpkinscm** [comment]
It isn't a full email client, it's just using the javax.mail classes to send and receive messages.  As for the coding, I don't see what you mean by having to re-code for every setup.  Isn't that the point of standardized protocols like IMAP?

**Comment by Mattwhooo** [comment]
I must not understand the setup. How do you receive emails with out running
a server? Are we talking about hooking in to an existing customer email
server as most of those would be very restricted for what can connect to it.

**Comment by simpkinscm** [comment]
Yes, it would be connecting to an existing client email server.  How exactly do you mean they would be restricted?

**Comment by JustinPhlegar** [comment]
The intent for this is to grant the ability to an read from email server, weather it be a clients server or to something like mail trap or gmail. It is common for an app to send an email notifications and validating that email maybe a requirement.

**Comment by Mattwhooo** [comment]
I'm likely just making stuff up here. I know there are restrictions for connecting mobile devices to these services, but maybe no when doing it in code.  Having said that it seems likely to me that these enterprise email servers would not allow just any code to connect to them with out some robust authentication and the ability to remote wipe.  Doing this against gmail, mailtrap or something like that makes sense to me though.

**Comment by simpkinscm** [comment]
The only email service I've found so far that I wasn't able to connect to was sending email with MS Exchange servers that don't have SMTP enabled.  For companies where this is the case, we can always create an account at the mail provider of their choosing if we need to send emails.
